networkmanager/networkmanager_test.go

```diff
@@ -71,542 +99,544 @@ func TestMain(m *testing.M) {
-	t.Cleanup(func() {
-		_ = manager.DeleteAllNetworks()
-	})
+	}()
 
-	numNetworks := 3
-	numServices := 3
+	cniConfig := cniConfig{}
 
-	for j := 0; j < numNetworks; j++ {
-		for i := 0; i < numServices; i++ {
-			if err := manager.AddServiceToNetwork(fmt.Sprintf("service%d%d", j, i),
-				fmt.Sprintf("network%d", j), networkmanager.NetworkParams{}); err != nil {
-				t.Fatalf("Can't add service to network: %s", err)
-			}
-		}
+	// need to test cleaning networking at startup
+	networkDir := path.Join(tmpDir, "cni", "networks", "network0", instanceID)
+	if err := os.MkdirAll(networkDir, 0o755); err != nil {
+		t.Fatal(err)
 	}
 
-	if err := manager.DeleteAllNetworks(); err != nil {
-		t.Fatalf("Can't delete all networks: %s", err)
+	manager, err := networkmanager.New(&config.Config{WorkingDir: tmpDir}, &cniConfig, nil)
+	if err != nil {
+		t.Fatal(err)
 	}
 
-	for j := 0; j < numNetworks; j++ {
-		for i := 0; i < numServices; i++ {
-			if err := manager.IsServiceInNetwork(fmt.Sprintf("service%d%d", j, i),
-				fmt.Sprintf("network%d", j)); err == nil {
-				t.Fatalf("Service should not be in network: %s", err)
-			}
-		}
-	}
-}
+	defer manager.Close()
 
-func TestInternet(t *testing.T) {
-	t.Cleanup(func() { _ = manager.DeleteNetwork("network0") })
+	networkmanager.GetIPSubnet = getIPSubnet
 
-	containerPath := path.Join(tmpDir, "servicenm1")
+	plugins := createPlugins([]string{createBridgePlugin(tmpDir + `/`), createDNSPlugin()})
 
-	if err := createOCIContainer(containerPath, "servicenm1", []string{"ping", "google.com", "-c10", "-w15"}); err != nil {
-		t.Fatalf("Can't create service container: %s", err)
+	if _, err := manager.GetInstanceIP(instanceID, "network0"); err == nil {
+		t.Fatal("Instance IP must not be present")
 	}
 
-	if err := manager.AddServiceToNetwork("servicenm1", "network0", networkmanager.NetworkParams{
-		ResolvConfFilePath: path.Join(containerPath, "etc", "resolv.conf"),
+	hostsPath := path.Join(container0Path, "hosts")
+	resolvConfPath := path.Join(container0Path, "resolv.conf")
+
+	if err := manager.AddInstanceToNetwork(instanceID, "network0", networkmanager.NetworkParams{
+		Hostname:           "myhost",
+		HostsFilePath:      hostsPath,
+		ResolvConfFilePath: resolvConfPath,
 	}); err != nil {
-		t.Fatalf("Can't add service to network: %s", err)
+		t.Fatalf("Can't add instance to network: %s", err)
 	}
 
-	if err := runOCIContainer(containerPath, "servicenm1"); err != nil {
-		t.Errorf("Error: %s", err)
+	content, err := readFromFile(hostsPath)
+	if err != nil {
+		t.Fatal(err)
 	}
-}
 
-func TestInterServiceConnection(t *testing.T) {
-	t.Cleanup(func() {
-		_ = killOCIContainer("servicenm2")
-		_ = manager.DeleteNetwork("network0")
-	})
-
-	container0Path := path.Join(tmpDir, "servicenm2")
-
-	if err := createOCIContainer(container0Path, "servicenm2", []string{"sleep", "infinity"}); err != nil {
-		t.Fatalf("Can't create service container: %s", err)
+	if content != createContentHostsFile("myhost") {
```

Comment:
Check expected string directly. It is not so big.

Reply:
Done